src/components/Order/Stepper.jsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import Box from "@mui/material/Box";
+import Stepper from "@mui/material/Stepper";
+import Step from "@mui/material/Step";
+import StepLabel from "@mui/material/StepLabel";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import useStep from "./useStep";
+import {containerStyle,stepContentBox,buttonContainer,
+         spacer,backButton,completeText} from "./Stepper.styles";
+
+import OrderSummary from "./steps/OrderSummary";
+import CustomerDetails from "./steps/CustomerDetails";
+import PaymentDetails from "./steps/PaymentDetails";
+import OrderComplete from "./steps/OrderComplete";
+
+const stepComponents = [
+    <OrderSummary />,
+    <CustomerDetails />,
+    <PaymentDetails />,
+    <OrderComplete />
+  ];
+  
+const renderStepContent = (step) => stepComponents[step] || null;
+
+const steps = [
+  "Order Summary",
+  "Customer Details",
+  "Payment Details",
+  "Order Complete"
+];
+
+export default function HorizontalLinearStepper() {
+  const { activeStep, next, back } = useStep(steps.length);
+
+  return (
+    <Box sx={containerStyle}>
+      <Stepper activeStep={activeStep}>
+        {steps.map((label) => (
+          <Step key={label}>
+            <StepLabel>{label}</StepLabel>
+          </Step>
+        ))}
+      </Stepper>
+
+      <Box sx={stepContentBox}>{renderStepContent(activeStep)}</Box>
+
+      {activeStep === steps.length-1 ? (
+         <Box sx={buttonContainer}>
+         <Box sx={spacer} />
+         <Button onClick={next}>Go Home</Button>
+       </Box>
+      ) : (
+        <>
+          <Box sx={buttonContainer}>
+            <Button
+              color="inherit"
+              disabled={activeStep === 0 || activeStep === steps.length - 1}
+              onClick={back}
+              sx={backButton}
+            >
```

Comment:
Save this `activeStep === steps.length - 1` into a variable because you are using this more than once. 
Also dont give `color='inherit'` - do it via the styled object (backButton).